girder_worker/docker/tasks/__init__.py

```diff
@@ -0,0 +1,304 @@
+        for read_stream_connector in read_stream_connectors:
+            if isinstance(read_stream_connector.input, ContainerStdErr):
+                stderr_reader = FileDescriptorReader(stderr.fileno())
+                read_stream_connector.output = DockerStreamPushAdapter(read_stream_connector.output)
+                read_stream_connector.input = stderr_reader
+                stderr_connected = True
+                break
+
+        # If not stdout and stderr connection has been provided just use
+        # sys.stdXXX
+        if not stdout_connected:
+            stdout_reader = FileDescriptorReader(stdout.fileno())
+            connector = ReadStreamConnector(
+                stdout_reader,
+                DockerStreamPushAdapter(StdStreamWriter(sys.stdout)))
+            read_stream_connectors.append(connector)
+
+        if not stderr_connected:
+            stderr_reader = FileDescriptorReader(stderr.fileno())
+            connector = ReadStreamConnector(
+                stderr_reader,
+                DockerStreamPushAdapter(StdStreamWriter(sys.stderr)))
+            read_stream_connectors.append(connector)
+
+        # Run select loop
+        utils.select_loop(exit_condition=exit_condition,
+                          readers=read_stream_connectors,
+                          writers=write_stream_connectors)
+
+        if task.canceled:
+            try:
+                container.stop()
+            # Catch the ReadTimeout from requests and wait for container to
+            # exit. See https://github.com/docker/docker-py/issues/1374 for
+            # more details.
+            except ReadTimeout:
+                tries = 10
+                while tries > 0:
+                    container.reload()
+                    if container.status == 'exited':
+                        break
+
+                if container.status != 'exited':
+                    msg = 'Unable to stop container: %s' % container.id
+                    logger.error(msg)
+            except DockerException as dex:
+                logger.error(dex)
+                raise
+
+    finally:
+        # Close our stdout and stderr sockets
+        if stdout:
+            stdout.close()
+        if stderr:
+            stderr.close()
+
+
+def _handle_streaming_args(args):
+    processed_arg = []
+    write_streams = []
+    read_streams = []
+
+    for arg in args:
+        if isinstance(arg, StreamConnector):
+            if isinstance(arg, WriteStreamConnector):
+                write_streams.append(arg)
+            else:
+                read_streams.append(arg)
+
+            # Get any container argument associated with this stream
+            arg = arg.container_arg()
+
+        processed_arg.append(arg)
+
+    return (processed_arg, read_streams, write_streams)
+
+
+class _RequestDefaultTemporaryVolume(_TemporaryVolumeBase):
+    def __init__(self):
+        super(_RequestDefaultTemporaryVolume, self).__init__(None, None)
+        self._dir = dir
```

Comment:
I'm guessing you didn't meant to point this at the builtin `dir` function

Reply:
Nope!